src/eos-crash-metrics.c

```diff
@@ -137,6 +178,91 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static char *
+get_associated_runtime (FlatpakInstalledRef *ref, GError **error)
+{
+  g_autoptr(GBytes) metadata = NULL;
+  gchar *runtime = NULL;
+  g_autoptr(GKeyFile) key_file = NULL;
+
+  metadata = flatpak_installed_ref_load_metadata (ref, NULL, error);
+  if (metadata == NULL)
+    return NULL;
+
+  key_file = g_key_file_new ();
+  if (!g_key_file_load_from_bytes (key_file, metadata, G_KEY_FILE_NONE, error))
+    return NULL;
+
+  if (!(runtime = g_key_file_get_string (key_file, "Application", "runtime", error)))
+    return NULL;
+
+  return runtime;
+}
+
+static FlatpakInfo *
+get_flatpak_info (const char *path,
+                  GError **error)
+{
+  FlatpakInstalledRef *app = NULL;
+  g_autoptr(GPtrArray) xrefs = NULL;
+  guint i;
+  g_autofree char *executable_name = NULL;
+  g_autoptr(FlatpakInstallation) installation = flatpak_installation_new_system (NULL, error);
+
+  if (installation == NULL)
+    return NULL;
+
+  executable_name = g_path_get_basename (path);
+
+  /* we are only interested in apps */
+  xrefs = flatpak_installation_list_installed_refs_by_kind (installation,
+                                                            FLATPAK_REF_KIND_APP,
+                                                            NULL, error);
+  if (xrefs == NULL)
+    return NULL;
+
+  for (i = 0; i < xrefs->len; i++)
+    {
+      g_autofree gchar *executable_path = NULL;
+      FlatpakInstalledRef *xref = g_ptr_array_index (xrefs, i);
+
+      executable_path = g_build_filename (flatpak_installed_ref_get_deploy_dir (xref),
+                                          "files",
+                                          "bin",
+                                          executable_name,
+                                          NULL);
+      /* found a Flatpak with the same application name */
+      if (g_file_test (executable_path, G_FILE_TEST_EXISTS))
+        {
+          app = xref;
+          break;
+        }
+    }
+
+  if (app == NULL)
+    return NULL;
+
+  g_autofree const char *runtime_name = get_associated_runtime (app, error);
+  if (runtime_name == NULL)
+    return NULL;
+
+  g_auto(GStrv) parts = g_strsplit (runtime_name, "/", 3);
+  if (g_strv_length (parts) != 3)
+    return NULL;
```

Comment:
Here you're returning `NULL` without setting the error

Reply:
Yes, and that would segfault with the new error handling code when we try to print out error->message. We actually had another occasion above when no app was found, fixed that one as well and set an error.